commit e1456788c48c5ed6b55bc107a7342d63f2a08413
Author: Matthew Gregan <kinetik@flim.org>
Date:   Sat Jul 24 23:37:17 2021 +1200

    wasapi: Avoid a few copies in `wasapi_find_matching_output_device`.
    
    See BMO #1722073.

diff --git a/src/cubeb_wasapi.cpp b/src/cubeb_wasapi.cpp
index f6f7619..22a095a 100644
--- a/src/cubeb_wasapi.cpp
+++ b/src/cubeb_wasapi.cpp
@@ -2263,15 +2263,14 @@ void wasapi_find_matching_output_device(cubeb_stream * stm) {
   // Find the input device, and then find the output device with the same group
   // id and the same rate.
   for (uint32_t i = 0; i < collection.count; i++) {
-    cubeb_device_info dev = collection.device[i];
-    if (dev.devid == input_device_id) {
-      input_device = &dev;
+    if (collection.device[i].devid == input_device_id) {
+      input_device = &collection.device[i];
       break;
     }
   }
 
   for (uint32_t i = 0; i < collection.count; i++) {
-    cubeb_device_info dev = collection.device[i];
+    cubeb_device_info & dev = collection.device[i];
     if (dev.type == CUBEB_DEVICE_TYPE_OUTPUT && dev.group_id && input_device &&
         !strcmp(dev.group_id, input_device->group_id) &&
         dev.default_rate == input_device->default_rate) {
